picard/acoustid/json_helpers.py

```diff
@@ -21,128 +21,125 @@
+    for rg in recording.get('releasegroups', []):
+        # Local caching: access once, use many times
+        rg_id = rg.get('id')
+        rg_type = rg.get('type')
+        rg_sec_types = rg.get('secondarytypes')
+        rg_title = rg.get('title')
+
+        for rel in rg.get('releases', []):
+            # Pre-build the base release structure
+            base_release = {
+                'id': rel.get('id'),
+                'release-group': {
+                    'id': rg_id,
+                    'primary-type': rg_type,
+                    'secondary-types': rg_sec_types
+                },
+                'title': rel.get('title', rg_title),
+                'country': rel.get('country'),
+                'date': rel.get('date'),
+                'medium-count': rel.get('medium_count'),
+                'track-count': rel.get('track_count'),
+                'media': []
+            }
+
+            # Optimization: Remove 'None' values from release-group in one pass
+            base_release['release-group'] = {
+                k: v for k, v in base_release['release-group'].items() if v is not None
+            }
+
+            for medium in rel.get('mediums', []):
+                media_mb = {
+                    'format': medium.get('format'),
+                    'track-count': medium.get('track_count'),
+                    'position': medium.get('position'),
+                }
                 if 'tracks' in medium:
-                    media_mb['track'] = medium['tracks']
-                    for track_mb in media_mb['track']:
-                        track_mb['number'] = track_mb['position']
-
-                release_mb['media'].append(media_mb)
-
-            # AcoustId service is returning country/date as attrib of the release, but really, according to MusicBrainz database schema definition,
-            # https://musicbrainz.org/doc/MusicBrainz_Database/Schema
-            # Its a one-to-many sub attribute in release events.
-            # They do return the releaseevents, but seem to be copying the first one on to the release.
-            # So if we have releaseevents, then use them to create multiple records, and ignore what is coming on the release
-            if 'releaseevents' in release:
-                for releaseevent in release['releaseevents']:
-                    release_mb['country'] = releaseevent.get('country', '')
-                    release_mb['date'] = releaseevent.get('date', '')
-                    release_list.append(release_mb)
+                    tracks = medium['tracks']
+                    for t in tracks:
+                        # Direct assignment is faster than calling functions repeatedly
+                        t['number'] = t.get('position')
+                    media_mb['track'] = tracks
+                
+                base_release['media'].append(media_mb)
+
+            # Fix the reference bug: append copies if there are multiple events
```

Comment:
Please make 2 small PRs:

- one for the fix, and create a matching ticket for the bug
- one for optimization

As is the PR is hard to review, too much changes at once.
Also you removed important comments.